sys/auto_init/saul/auto_init_bme280.c

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (C) 2016 Kees Bakker, SODAQ
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     auto_init_saul
+ * @{
+ *
+ * @file
+ * @brief       Auto initialization of BME280 driver.
+ *
+ * @author      Kees Bakker <kees@sodaq.com>
+ *
+ * @}
+ */
+
+#ifdef MODULE_BME280
+
+#include "log.h"
+#include "saul_reg.h"
+
+#include "bme280_params.h"
+#include "bme280.h"
+
+/**
+ * @brief   Define the number of configured sensors
+ */
+#define BME280_NUMOF    (sizeof(bme280_params) / sizeof(bme280_params[0]))
+
+/**
+ * @brief   Allocation of memory for device descriptors
+ */
+static bme280_t bme280_devs[BME280_NUMOF];
+
+/**
+ * @brief   Memory for the SAUL registry entries
+ */
+static saul_reg_t saul_entries[BME280_NUMOF * 3];
+
+void auto_init_bme280(void)
+{
+    size_t se_ix = 0;
```

Comment:
I don't think this is needed. You can use the loop variable `i` with i for temperature, i+1 for humidity, and i+2 for pressure.